### PR TITLE
core: fix backlight flickering at the end of PIN loader

### LINKED ISSUE #3311: Non-fluent loaders
**Describe the bug**
Buggy loaders, see video below.

**Firmware version and revision**
2.6.2 (debug and normal)

**Expected behavior**
A fluent loader.

https://github.com/trezor/trezor-firmware/assets/57008159/017f8723-b9ab-44bf-b21e-96c692510e7a

### PR DESCRIPTION
**Pull request by mmilata**
This bug was introduced by @mmilata in #2593 and possibly was made more visible by dee559f4f9ea61835e8a97bee541f6ce5edab121. Currently it is causing 3x fadeout+fadein just before the PIN progress finishes. Please test on device because emulator does not show backlight behavior well. Fixes #3311.

[UI diff](https://satoshilabs.gitlab.io/-/trezor/trezor-firmware/-/jobs/5303430625/artifacts/master_diff/index.html)

### CODE REVIEW

**Comment by Hannsek** [comment]
Loader are not flickering anymore. @trezor/qa please test it 🙏🏻

File: core/src/trezor/pin.py [hunk 1]
```diff
@@ -80,7 +80,7 @@ def show_pin_timeout(seconds: int, progress: int, message: str) -> bool:
     _progress_layout.report(progress, remaining)
 
     # drop the layout when done so trezor.ui doesn't have to remain in memory
-    if seconds == 0:
+    if progress == 1000:
```
Comment (matejcik) [inline core/src/trezor/pin.py:83]: to be safe
```suggestion
    if progress >= 1000:
```

**Review by grdddj — APPROVED** [review]
LGTM, thumbs up to Matejcik's suggestion

**Comment by mmilata** [comment]
Changelog check failing because of #3342.

**Comment by bosomt** [comment]
**QA OK**

**Info:**
 - Suite version: desktop 23.11.0 (40b035b2be45a4792adc9fb8373fdacff7b89bb3)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuiteDev/23.11.0 Chrome/116.0.5845.188 Electron/26.2.1 Safari/537.36
 - OS: MacIntel
 - Screen: 1440x900
 - Device: Trezor T2T1 2.6.3 regular (revision b95b2182fa67f7823f25e724d6702e531312e918)
 - Transport: BridgeTransport 2.0.33

**Pull request merged**